### PR TITLE
boost: update to 1.79.0.

### PR DESCRIPTION
**Pull request by JamiKettunen**
[ci skip]

#### Packages to fix (from `xrevshlib boost` as locally build-tested on `x86_64-musl`)
- [x] Aegisub (fails on `do_build()`) - awaiting new upstream release, tip of `dev` branch builds
- [x] PrusaSlicer (fails on `do_build()`) - backported upstream boost 1.79 compile fixes to v2.4.2
- [ ] TSC (fails on `do_build()`)
- [ ] cryfs (fails on `do_build()`)
- [ ] fifengine (fails on `do_build()`)
- [x] uhd (fails on `do_build()`)
  - [ ] gnuradio (failed on `do_build()`, updating to [v3.10.2.0](https://github.com/gnuradio/gnuradio/releases/tag/v3.10.2.0) appears to have disabled building a lot of components including all shared objects and pkg-config files for example, probably just a bunch of missing new "optional" build-time dependencies...)
    - [ ] gnuradio-osmosdr (depends on `gnuradio`)
- [ ] imagescan (fails on `do_build()`)
- [ ] kea (fails on `do_build()`)
- [ ] ledger (fails on `do_build()`)
- [ ] libbitcoin-system (fails on `do_build()`)
  - [ ] libbitcoin-database (depends on `libbitcoin-system`)
    - [ ] libbitcoin-blockchain (depends on `libbitcoin-database`)
      - [ ] libbitcoin-node (depends on `libbitcoin-blockchain`)
      - [ ] libbitcoin-server (depends on `libbitcoin-blockchain`)
  - [ ] libbitcoin-protocol (depends on `libbitcoin-system`)
    - [ ] libbitcoin-client (depends on `libbitcoin-protocol`)
      - [ ] libbitcoin-explorer (depends on `libbitcoin-client`)
  - [ ] libbitcoin-network (depends on `libbitcoin-system`)
- [ ] libetonyek (fails on `do_build()`, appears to need `-std=c++17` or higher? `NUM3Parser.cpp:46:8: error: 'for_each' is not a member of 'std'`)
  - [ ] libreoffice (depends on `libetonyek`)
- [ ] librime (fails on `do_build()`)
- [ ] namecoin (fails on `do_build()`)
- [ ] pulseeffects-legacy (fails on `do_build()`)
- [ ] supercollider (fails on `do_build()`)
- [ ] leatherman (hangs on `do_check()`, test process pinned one core at 100% for 10 minutes)
  - [x] facter (depends on `leatherman`)
- [ ] inkscape (fails on `do_check()`)
- [ ] libtorrent-rasterbar (hangs on `do_check()`, 90/97 test_web_seed_http stuck for 10 minutes)
- [ ] schroot (fails on `do_check()`)
- [ ] bitcoin (fails on `do_check()`)

#### Packages to fix (from `xrevshlib uhd` as locally build-tested on `x86_64-musl`)
- [ ] urh (fails on `do_check()` even after adding `checkdepends="python3-pip"`)

#### Modified patches (boost)
`exec_file.patch`: Removed due to being upstream as https://github.com/boostorg/python/commit/d9f06052e28873037db7f98629bce72182a42410
`mips.patch`: Updated to apply on top of https://github.com/boostorg/fiber/commit/5dc9dabc5d5ff8da53ec897c265c9cba1b6ed0df, not sure if this can be removed now(?)

#### Misc changes
- I've renamed all subpackages of `boost` so that the versioned name doesn't conflict with other numbers and are consistent with the previously existing `boost-python3-1.72`, e.g. `libboost_math_c991.72` is now `libboost_math_c99-1.79` (notice the `-`!)

#### Testing the changes
- I tested the changes in this PR: **NO**

### CODE REVIEW

File: srcpkgs/boost/template [hunk 1]
```diff
@@ -31,24 +32,19 @@ replaces="libboost_program_options1.69>=0 libboost_math_c99f1.69>=0
...
+ libboost_graph1.79 libboost_locale1.79 libboost_regex1.79
+ libboost_type_erasure1.79 libboost_coroutine1.79
+ libboost_atomic1.79 libboost_prg_exec_monitor1.79 libboost_math_c991.79
+ boost-python3-1.79 libboost_log_setup1.79 boost-devel
+ libboost_timer1.79 libboost_system1.79 libboost_unit_test_framework1.79
+ libboost_serialization1.79 libboost_math_tr11.79 boost-build
+ libboost_math_c99l1.79 boost-python1.79 libboost_contract1.79
+ libboost_date_time1.79 libboost_log1.79 libboost_random1.79
+ libboost_thread1.79 libboost_container1.79 boost-jam
+ libboost_wserialization1.79 libboost_context1.79"
```
Comment (ahesford) [inline srcpkgs/boost/template:47]: If you're dropping the no-cross `libboost_stacktrace_addr2line1.72`,  you should be able to just drop the entier `subpackages` declaration.

Comment (Johnnynator) [reply srcpkgs/boost/template:47]: subpackages is currently abused  in `boost-devel` as a list of deps, so you would have to move the list of packages to there.

**Comment by q66** [comment]
It's definitely not okay to remove the subpackages, it seems this is a bug in boost. The libraries are being built, but only their headers and cmake files get installed.

**Comment by atweiden** [comment]
> cryfs (fails on `do_build()`)

cryfs builds succeed with [boost-1.79.0](https://github.com/atweiden/voidpkgs/tree/cryfs-boost1.79.0)

**Comment by classabbyamp** [comment]
can we look into removing the python2 boost subpkg? it seems there's currently nothing using it

**Comment by JamiKettunen** [comment]
Closing in favor of #38949.

**Pull request closed without merging**